Sources/GoodPersistence/KeychainValue.swift

```diff
@@ -25,63 +74,137 @@ public class KeychainValue<T: Codable> {
     }
 
     private let subject: PassthroughSubject<T, Never> = PassthroughSubject()
+    private let newSubject: PassthroughSubject<T, KeychainError> = PassthroughSubject()
+    
     private let key: String
     private let defaultValue: T
-    private let accessibility: KeychainItemAccessibility?
+    private let accessibility: KeychainAccess.Accessibility?
+    private let synchronizable: Bool
+    private let authenticationPolicy: KeychainAccess.AuthenticationPolicy?
 
     /// Initializes a KeychainValue instance with a given key, default value, and accessibility.
     /// - Parameters:
     ///   - key: The key for the Keychain item
     ///   - defaultValue: The default value for the Keychain item
     ///   - accessibility: The accessibility level for the Keychain item. The default value is nil.
-    public init(_ key: String, defaultValue: T, accessibility: KeychainItemAccessibility? = nil) {
+    public init(
+        _ key: String,
+        defaultValue: T,
+        accessibility: KeychainAccess.Accessibility? = nil,
+        synchronizable: Bool = false,
+        authenticationPolicy: KeychainAccess.AuthenticationPolicy? = nil
+    ) {
         self.key = key
         self.defaultValue = defaultValue
         self.accessibility = accessibility
+        self.synchronizable = synchronizable
+        self.authenticationPolicy = authenticationPolicy
     }
     
     /// Provide the wrapped value to the user which is retrieved from Keychain or the default value
     public var wrappedValue: T {
         get {
-            // Retrieve data from Keychain using the key and accessibility if specified
-            guard let data = KeychainWrapper.standard.data(
-                forKey: key,
-                withAccessibility: accessibility
-            ) else {
-                // Return default value if data cannot be retrieved from Keychain
+            let keychain = setupKeychain()
+            do {
+                // Retrieve data from Keychain using the key and accessibility if specified
```

Comment:
Edit tych komentaro este a pis o tom preco nieco robi nie co sa tam deje.

To co sa tam deje je jasne z kodu. To preco sa to tak robi je enigmou :)